quesma/queryparser/pancake_model.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package queryparser
+
+import (
+	"quesma/logger"
+	"quesma/model"
+)
+
+type pancakeAggregationTopLevel struct {
+	children    []*pancakeAggregationLevel
+	whereClause model.Expr
+}
+
+type pancakeAggregationLevel struct {
```

Comment:
`pancakeAggregationLevel` contains everything from `pancakeFillingMetricAggregation`, embedding?

Reply:
There is a process that transforms a tree of aggregation into a list of aggregation.

`pancakeAggregationLevel` is part of the input tree, `pancakeFillingMetricAggregation` is a part of the result.

I think we should keep them separate.

Reply:
ok, but from content perspective they store the same set of fields, so maybe we can extract common part and embed it